index.js

```diff
@@ -186,24 +187,37 @@ function getMetrics({ projectKey, startTime, endTime, prNum }) {
 }
 
 /**
- * Determine Sonar project key and username based on SonarQube edition
+ * Determine Sonar project key, project name, and username based on:
+ * - SonarQube edition
+ * - job annotation
  * @method getProjectData
  * @param  {Object}     config
+ * @param  {Object}     [config.annotations]        Screwdriver job annotations
  * @param  {Boolean}    config.enterpriseEnabled    If enterprise is enabled
- * @param  {String}     config.jobId                Job ID
- * @param  {String}     config.pipelineId           Pipeline ID
- * @return {Object}                                 Sonar project key and username
+ * @param  {String}     config.jobId                Screwdriver job ID
+ * @param  {String}     config.jobName              Screwdriver job name
+ * @param  {String}     config.pipelineId           Screwdriver pipeline ID
+ * @param  {String}     config.pipelineName         Screwdriver pipeline name
+ * @return {Object}                                 Sonar project key, project name, and username
  */
-function getProjectData({ enterpriseEnabled, jobId, pipelineId }) {
-    if (enterpriseEnabled) {
+function getProjectData({ annotations, enterpriseEnabled, jobId, jobName, pipelineId,
+    pipelineName }) {
+    // Figure out default scope: pipeline scope for enterprise edition, job scope for everything else
+    const defaultScope = enterpriseEnabled ? 'pipeline' : 'job';
+    // Use user-configured scope or default scope
+    const scope = annotations ? annotations[COVERAGE_SCOPE_ANNOTATION] : defaultScope;
```

Comment:
this should be ```annotations && annotations[COVERAGE_SCOPE_ANNOTATION]``` else won't scope get's overwritten to `null` if it doesn't exist in `annotations` ?

Reply:
updated